dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java

```diff
@@ -75,17 +75,27 @@ public String getFinalSqlExpression(String tableLoc) {
     } else if ( this.getSqlExpression() != null) {
       return this.getSqlExpression();
     } else {
-      String [] splitIndex = this.getMultiFieldNames().split(",");
+      String [] splitIndex = this.getMultiFieldNames().split(" *, *");
 
-      String result = "concat_space_sql(";
+      StringBuilder result = new StringBuilder("concat_space_sql(");
       for(int i = 0;i < splitIndex.length;i++) {
         if(i != 0) {
-          result += " , ";
+          result .append(" , ");
         }
-        result += tableLoc + ".jsonb->>'" + splitIndex[i] + "'";
+        String [] rawExpandedTerm = splitIndex[i].split("\\.");
+        StringBuilder expandedTerm = new StringBuilder();
+
+        for(int j = 0; j < rawExpandedTerm.length; j++) {
+          String arrowToken = "->";
+          if(j == rawExpandedTerm.length - 1) {
+            arrowToken = "->>";
+          }
+          expandedTerm.append(arrowToken).append("'").append(rawExpandedTerm[j]).append("'");
+        }
+        result.append(tableLoc).append(".jsonb").append(expandedTerm);
```

Comment:
Calculating expandedTerm should be extracted into a separate method.
Using StringBuilder is a good idea. This can be extended by avoiding .split("\\.") that creates Strings.

Example:

```
  /**
   * Append the SQL JSONB field lookup to sb. Examples: For "a" append "->>'a'",
   * for "a.b" append "->'a'->>'b'", for "foo.bar.baz" append "->'foo'->'bar'->>'baz'".
   * @param sb  where to append.
   * @param field  the dot separated path to the field
   */
  static void appendSqlTextField(StringBuilder sb, String field) {
    int pos = 0;
    int lastDot = field.lastIndexOf('.');
    int nextDot;
    while (true) {
      nextDot = field.indexOf('.', pos);
      if (pos > lastDot) {
        break;
      }
      sb.append("->'").append(field, pos, nextDot).append("'");
      pos = nextDot + 1;
    }
    sb.append("->>'").append(field, pos, field.length()).append("'");
  }
```